### ISSUE TITLE
Research Product Mapping

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#34847]
# Description
We need to address several key questions related to developing a mapping from Quality Assurance practice artifacts to products hosted on VA.gov. Key questions are listed below --

- What is the definition of a product?
- Can we create a machine parse-able schema for defining a product?
  - How do we do this for products in `vets-website`?
  - How do we do this for products in `vets-api`?
- What additional do we need to collect in order to develop the mapping?

[Mural exploration](https://app.mural.co/t/adhocvetsgov9623/m/adhocvetsgov9623/1642092284056/686df56755a1830c547782e2ae95433b90c821ec?sender=peterhill4315) of data sources mapping to a unified product definition.

## Acceptance Criteria
Our proposed solution -- 
- [x] Standardizes the definition of a product on VA.gov
- [x] Allows us to automatically retrieve details of that product
- [x] Enables the team to automatically map the product to the QA practice output for the product as listed below
  - [x] E2E testing can be mapped to the product definition
  - [x] FE unit testing can be mapped to the product definition
  - [x] BE unit testing can be mapped to the product definition
  - [x] Contract tests can be mapped to the product definition
  - [x] Manual tests can be mapped to the product definition
  - [x] Load tests can be mapped to the product definition
  - [x] Synthetic monitors can be mapped to the product definition

**Comment by pjhill** [comment]
Our team brainstorming session provided the following notes and feedback --

- A product is a behavior or a collection of related behaviors hosted on VA.gov that accomplish a business goal.
- A product is whatever the VFS team defines as the scope of their work.
- A product is a collection of code located in the same directory.
- A product is the code in a directory in src/applications in vets-website
- Are there different 'types' of products?
- Should different types of products be treated differently regarding what are 'key metrics'?
- The [Product Directory](https://vfs.atlassian.net/wiki/spaces/VAProducts/overview) doesn't contain a ton of useful information to link QA artifacts other than a potential "master list" of products
- Q: What is desired output for our team for the Product definition / directory / master reference data?
  - A: Output should be a list of human readable names for products.
- Q: Is it worth mining the existing Product Directory for the information that we need given the answer to ^ 
  - A: The Product Directory does not contain an easy way to map to QA artifacts
- A manifest.json following a schema is required for each product.
- The manifest would not solve the problem of tracking anything outside of vets-website.
- We could have product information contained in manifest that is not relevant to vets-website.
- Not all products have manifest currently.
- Manifest is not autogenerated
- Do all products have a front-end hosted in vets-website?
- If so, can we enforce the creation of a manifest as a requirement for participation in vets-website. 
- What if Platform / QA standards created a manifest for each product?
- Would teams disagree with the information in the manifest?
- What if a team changes a manifest without following the schema we assume for ingest?
- Discuss a potential for a 'manifest' like file in vets-api with BE team.
- Caveat -- GitHub is organized by team and not product. Mapping from team to product with only a team label available is... impossible???
- Bottom line -- We need to enforce standards about manifests and/or product definition.
- Could we use a linter to enforce schema adherence with manifests?
- What happens if a product renames???
- People break data linking by altering fundamental attributes of product.
- UID for products that is immutable.
- vets-api back-end tests has multiple products combined.
- models and controllers are shared resources between multiple products.
- No 1:1 in vets-api
- How to deal with challenge of vets-api's inherent 1:many product code distribution?
- Are all products in vets-website src?
  - A: Yes....???
- Does Contract testing provide any insight as to the mapping of front-end to back-end?

**Comment by JoeTice** [comment]
Discussion and brainstorming sessions around this task has produced a few task-level action items that don't fit under this epic, and we don't feel there is further work or research to be done on the matter if these tasks prove fruitful...

[Brainstorming Session 1
](https://app.mural.co/t/adhocvetsgov9623/m/adhocvetsgov9623/1641919554629/7193f97680215e7ee3f699e7be8f02d8109eb6f7?sender=uc4d00ea50beebdfad05a4846)

[Brainstorming Session 2](https://app.mural.co/t/adhocvetsgov9623/m/adhocvetsgov9623/1642092284056/686df56755a1830c547782e2ae95433b90c821ec?sender=uc4d00ea50beebdfad05a4846)

Closing this task
